File: loops/notes.py
# Gabriel Lopez, Loops Notes on Python

#1.What is a loop? 
#A section of code that repeats multiple times

#2. What are the two types of loops?
# for loop
nums = [12,3,66,7,3,3,2]

for num in nums:
    print(num)

    #while loop
x = 0

while x < 10:
    print(x)
    x+= 1 

#3. What is iteration
# That specific instance of the loop
# iteration the amount of times you are looping through 

#4. What are lists?
# A list is a group of many things that are related. IN PYTHON, a list is a variable that stores multiple values
nums = [1,2,3,4,5,6,7,6]
siblings = ["Alex", "Katie", "Andrew", "Vienna", "Tia", "Treyson", "Xavier", "Hailey"]
print(nums) #prints whole list is ugly
print(siblings[3]) #prinys 1 item from the list

siblings[7] = "Jake"
siblings.pop(5)
siblings.insert(1, "Jayshree")
siblings.insert(2, ["Joe","Noah", "Zee"])
print(siblings)


#5. How do you make lists in python? 
# Step 1. use brackets
# Step 2. add in items with CORRECT DATA TYPES. Strings have quotations marks, integers do not.
# Step 3. have a coma between each item.

#6. How do you make for loops in python? 
for sibling in siblings: #When you make a list, use a plural word. Then you put the singular version
    print(sibling)

for x in range(0, 101, 20):
    print(x) 

#7. How do you make while loops in python?
import random
x = 1 #varible to keep count of iteration
goose = random.randint(1,20)

while x <= 20:
    if x == goose:
        print("Goose!")
        break #tells the loop to stop
    else:
        print("Duck")
    x+= 1

#continue moves on to the next iteration without finishing 

File: loops/notes.c
// Gabriel Lopez, Notes on C
#include <stdio.h>

int main(void){
// 1. What is a loop?
    //a section of code repeated multiple time

//2. What are the two types of loops?
    //While Loops
    int start = 0;
    while(start <5){
        printf("Hello!\n");
        start++;
    }
    //For Loops
    int q;
    for(q=0;q<5;q++){
        printf("%d\n", q);
    }

//3. What is iteration
    // Repeating something with minor changes each time


//4. What are arrays(list)?
    // An array is a variable holding multiple values

// 8. How do you make arrays(lists) in C?
    // Array items must all be the same data type!
int grades[] = {88, 97, 100, 70, 72, 99, 61};
    // 1. set data type
    // 2. AFTER naming put brackets and write the length of the list
    // 3. List is surrounded by curly brackets {}
    // 4. Commas , between each item
    // print a single item from a list
    printf("CSP Grade: %d\n", grades[2]);
    // Change an item in the array
    grades[2] = 95;
    printf("CSP Grade: %d\n", grades[2]);
    // Size of array(list) in bytes
    int size = sizeof(grades);
    // length in arrays(list) items
    int length = sizeof(grades)/sizeof(grades[0]);
    printf("%d\n", length); //Every item in the list is taking up four bites of space. Printing it will print out the amount in Bits
    //array with strings
    //First brackets sets length of array(list)
    //Second brakets sets length of each string
    char teams[][16] = {"Thunder", "Lakers", "Cavaliers", "Celtics", "Bucks", "Nuggets"};
    printf("%s\n are winning the championship this season", teams[3]);
    int tlength = sizeof(teams)/sizeof(teams[0]);

// 9. How do you make for loops in C?
    int x; // set the iterator, keeps track of times through the loop (best practice as set for x and i)
    // in parens (starting point; ending point; count by)
    for (x=0;x<=10;x+=2){
        printf("%d\n", x);
    }
    int m;
    for(m=0;m<tlength;m++){
        printf("%s\n", teams[m]);
    }

// 10. How do you make while loops in C?
    int w = 100;//start point
    while(w>=0){//stop point
        printf("%d\n", w);
        w -= 10;//What we count by
    }
    return 0;
}